delta/sourcing/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/processor/EventSourceProcessor.scala

```diff
@@ -115,115 +113,234 @@ private[processor] class EventSourceProcessor[State, Command, Event, Rejection](
     * The behavior of the underlying state actor when we opted for NOT persisting the events
     */
   private def transientBehavior(
-      t: TransientEventDefinition[State, Command, Event, Rejection]
-  ): Behavior[EventSourceCommand] = {
-    def behavior(state: State): Behaviors.Receive[EventSourceCommand] =
-      Behaviors.receive[EventSourceCommand] { (_, cmd) =>
+      t: TransientEventDefinition[State, Command, Event, Rejection],
+      parent: ActorRef[ProcessorCommand]
+  ): Behavior[ChildActorRequest] = {
+    def behavior(state: State): Behaviors.Receive[ChildActorRequest] =
+      Behaviors.receive[ChildActorRequest] { (_, cmd) =>
         cmd match {
-          case RequestState(_, replyTo: ActorRef[StateReply[_]])                     =>
-            replyTo ! StateReply(state)
+          case ChildActorRequest.RequestState(replyTo) =>
+            replyTo ! AggregateResponse.StateResponse(state)
             Behaviors.same
-          case RequestStateInternal(id, replyTo: ActorRef[ResponseStateInternal[_]]) =>
-            replyTo ! ResponseStateInternal(id, state)
+          case ChildActorRequest.RequestLastSeqNr(_)   =>
             Behaviors.same
-          case _: RequestLastSeqNr                                                   =>
+          case ChildActorRequest.RequestStateInternal  =>
+            parent ! ChildActorResponse.StateResponseInternal(state)
             Behaviors.same
-          case Append(id, Event(event), replyTo: ActorRef[AppendSuccess[_, _]])      =>
+          case ChildActorRequest.Append(Event(event))  =>
             val newState = t.next(state, event)
-            replyTo ! AppendSuccess(id, event, newState)
+            parent ! ChildActorResponse.AppendResult(event, newState)
             behavior(newState)
         }
       }
     behavior(t.initialState)
   }
 
   /**
-    * Behavior of the actor responsible for handling commands and events
+    * Behavior of the actor responsible for handling commands and events.
+    * There are multiple behaviours, each of them responsible for a different stage of a command evaluation.
+    *
+    * The following is a diagram of the different behaviors and their order:
+    *
+    * ┌--------┐        ┌-------------------┐        ┌------------┐         ┌-----------┐
+    * │        │        │                   │        │            │ !druRun │           │
+    * │ active +--------> waitingToEvaluate │--------> evaluating │---------> appending │---┐
+    * │        │        │                   │        │            │---┐     │           │   │
+    * └---^----┘        └-------------------┘        └------------┘   │     └-----------┘   │
+    *     │                                                           │                     │
+    *     └-----------------------------------------------------------┴---------------------┘
+    *                                        dryRun
```

Comment:
This is nice :)
!druRun => !dryRun
waitingToEvaluate => fetchingState ?
The error part is missing

Reply:
I made the names changed and fixed the typo. Not sure what you mean with the error part, but that's just the behaviors overview...I didn't want to add too much detail, or it makes it difficult to draw :)

Reply:
I was speaking about the evaluationError case